services/web/server/src/simcore_service_webserver/projects/_delete.py

```diff
@@ -0,0 +1,148 @@
+
+    NOTE: this does NOT use fire&forget anymore. This is a decision of the caller to make.
+
+    raises ProjectDeleteError
+    """
+
+    log.debug(
+        "Deleting project '%s' for user '%s' in database",
+        f"{project_uuid=}",
+        f"{user_id=}",
+    )
+    db: ProjectDBAPI = app[APP_PROJECT_DBAPI]
+
+    try:
+        mark_project_as_deleted(app, project_uuid)
+
+        # stops dynamic services
+        # - raises ProjectNotFoundError, UserNotFoundError, ProjectLockError
+        await remove_project_dynamic_services(
+            user_id, project_uuid, app, notify_users=False
+        )
+
+        # stops computational services
+        # - raises DirectorServiceError
+        await director_v2_api.delete_pipeline(app, user_id, UUID(project_uuid))
+
+        # rm data from storage
+        await delete_data_folders_of_project(app, project_uuid, user_id)
+
+        # rm project from database
+        await db.delete_user_project(user_id, project_uuid)
+
+    except ProjectLockError as err:
+        raise ProjectDeleteError(
+            project_uuid, reason=f"Project currently in use {err}"
+        ) from err
+
+    except (ProjectNotFoundError, UserNotFoundError) as err:
+        raise ProjectDeleteError(
+            project_uuid, reason=f"Invalid project state {err}"
+        ) from err
+
+
+def create_delete_project_background_task(
+    app: web.Application,
+    project_uuid: str,
+    user_id: int,
+    remove_project_dynamic_services: Callable,
+    logger: logging.Logger,
+) -> asyncio.Task:
+    """Wrap `delete_project` for a `project_uuid` and `user_id` into a Task and schedule its execution in the event loop.
+
+    Return the scheduled Task
+    """
+
+    def _log_errors(fut: asyncio.Future):
+        try:
+            fut.result()
+        except Exception:  # pylint: disable=broad-except
+            logger.exception(
+                f"Error while deleting {project_uuid=} owned by {user_id=}"
+            )
+
+    task = asyncio.create_task(
+        delete_project(app, project_uuid, user_id, remove_project_dynamic_services),
+        name=DELETE_PROJECT_TASK_NAME.format(project_uuid, user_id),
+    )
+
+    assert task.get_name() == DELETE_PROJECT_TASK_NAME.format(project_uuid, user_id)
+
+    task.add_done_callback(functools.partial(_log_errors, log))
+    return task
+
+
+def get_delete_project_background_tasks(
+    project_uuid: ProjectID, user_id: UserID
+) -> List[asyncio.Task]:
+    """Returns delete-project task if not yet finished"""
+    return [
+        task
+        for task in asyncio.all_tasks()
+        if task.get_name() == DELETE_PROJECT_TASK_NAME.format(project_uuid, user_id)
+    ]
+
+
+def is_delete_project_background_task_running(
+    project_uuid: ProjectID, user_id: UserID
+) -> bool:
+    tasks = get_delete_project_background_tasks(project_uuid, user_id)
+    assert len(tasks) <= 1
+    return len(tasks) > 0
+
+
+# alias
```

Comment:
Why these aliases below? Why not use them as function names.

Reply:
i had a dilemma with names: inside ``_delete`` the functions make sense with the word ``delete`` but outside, it makes it too long. At the end I have both. It is a matter of convenience and does not add a huge overhead. But if you think otherwise I can change that

Reply:
I would live it, if you like ti accept the suggested rename below.